firmware/main.c

```diff
@@ -157,36 +166,44 @@ static void writeWordToPageBuffer(uint16_t data) {
       data = OSCCAL;
    }     
 #endif
-  
   boot_page_fill(currentAddress.w, data);
   currentAddress.w += 2;
 }
 
 /* ------------------------------------------------------------------------ */
 static uint8_t usbFunctionSetup(uint8_t data[8]) {
   usbRequest_t *rq = (void *)data;
- 
+  
+  idlePolls.b[1]=0; // reset idle polls when we get usb traffic
```

Comment:
I noticed you moved around idlePolls resetting. What is the purpose of this?

Reply:
[searching, it's been a while...]

From my post on #132 on Nov 20, 2017:
> The other issue was that apparently the idle timer was reset on any USB communication, not only on communication with the command-line tool. So it got reset every time the OS was bugging the device with something

IIRC it kept the bootloader from starting the application program for as long as an active USB host is around.